
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>cryptnoxpy.exceptions &#8212; CryptnoxPy  documentation</title>
    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/alabaster.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/graphviz.css" />
    <script data-url_root="../../../" id="documentation_options" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
   
  <link rel="stylesheet" href="../../../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="module-cryptnoxpy.exceptions">
<span id="cryptnoxpy-exceptions"></span><h1><a class="reference internal" href="#module-cryptnoxpy.exceptions" title="cryptnoxpy.exceptions"><code class="xref py py-mod docutils literal notranslate"><span class="pre">cryptnoxpy.exceptions</span></code></a><a class="headerlink" href="#module-cryptnoxpy.exceptions" title="Permalink to this headline">¶</a></h1>
<p>Module containing all exceptions that CryptnoxPy module can raise.</p>
<section id="module-contents">
<h2>Module Contents<a class="headerlink" href="#module-contents" title="Permalink to this headline">¶</a></h2>
<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.CryptnoxException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">CryptnoxException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.CryptnoxException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-obj docutils literal notranslate"><span class="pre">Exception</span></code></p>
<p>Base exception for the class exceptions.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.CardClosedException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">CardClosedException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.CardClosedException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-obj docutils literal notranslate"><span class="pre">Exception</span></code></p>
<p>The card wasn’t opened with PIN code or challenge-response</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.CardException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">CardException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.CardException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>No card was detected in the card reader.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.CardTypeException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">CardTypeException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.CardTypeException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>The detected card is not supported by this library</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.CertificateException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">CertificateException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.CertificateException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>There was an issue with the certification</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.ConnectionException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">ConnectionException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.ConnectionException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>An issue occurred in the communication with the reader</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.DataException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">DataException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.DataException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>The reader returned an empty message.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.DataValidationException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">DataValidationException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.DataValidationException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>The sent data is not valid.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.DerivationSelectionException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">DerivationSelectionException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.DerivationSelectionException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Not a valid derivation selection.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.EOSKeyError">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">EOSKeyError</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.EOSKeyError" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>The signature wasn’t compatible with EOS standard after 10 tries</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.FirmwareException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">FirmwareException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.FirmwareException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>There is an issue with the firmware on the card</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.GenuineCheckException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">GenuineCheckException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.GenuineCheckException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>The detected card is not a genuine Cryptnox product.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.GenericException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">GenericException</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">status</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bytes</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#cryptnoxpy.exceptions.GenericException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Generic exception that can mean multiple things depending on the call to the card</p>
<p>Process stats and throw a specific Exception from it.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.InitializationException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">InitializationException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.InitializationException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>The card hasn’t been initialized.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.KeyAlreadyGenerated">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">KeyAlreadyGenerated</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.KeyAlreadyGenerated" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Key can not be generated twice.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.SeedException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">SeedException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.SeedException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Keys weren’t found on the card.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.KeyGenerationException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">KeyGenerationException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.KeyGenerationException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Error in key generation.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.PinAuthenticationException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">PinAuthenticationException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.PinAuthenticationException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Error in turning off PIN authentication. There is no user key in the card</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.PinException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">PinException</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">number_of_retries</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">message</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'Wrong</span> <span class="pre">pin'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#cryptnoxpy.exceptions.PinException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Sent PIN code is not valid.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>number_of_retries</strong> (<em>int</em>) – Number of retries to send the PIN code
before the card is locked.</p></li>
<li><p><strong>message</strong> (<em>str</em>) – Optional message</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.PukException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">PukException</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">number_of_retries</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">message</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'Wrong</span> <span class="pre">PUK'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#cryptnoxpy.exceptions.PukException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Sent PUK code is not valid.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>number_of_retries</strong> (<em>int</em>) – Number of retries to send the PIN code
before the card is locked.</p></li>
<li><p><strong>message</strong> (<em>str</em>) – Optional message</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.ReadPublicKeyException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">ReadPublicKeyException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.ReadPublicKeyException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Data received during public key reading is not valid.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.ReaderException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">ReaderException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.ReaderException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Card reader wasn’t found attached to the device.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.SecureChannelException">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">SecureChannelException</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.SecureChannelException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Secure channel couldn’t be established.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.SoftLock">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">SoftLock</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.SoftLock" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>The card is soft locked, and requires power cycle before it can be opened</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="cryptnoxpy.exceptions.CardNotBlocked">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">cryptnoxpy.exceptions.</span></span><span class="sig-name descname"><span class="pre">CardNotBlocked</span></span><a class="headerlink" href="#cryptnoxpy.exceptions.CardNotBlocked" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#cryptnoxpy.exceptions.CryptnoxException" title="cryptnoxpy.exceptions.CryptnoxException"><code class="xref py py-obj docutils literal notranslate"><span class="pre">CryptnoxException</span></code></a></p>
<p>Trying to unlock unblocked card</p>
</dd></dl>

</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../../../index.html">CryptnoxPy</a></h1>








<h3>Navigation</h3>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../../../index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2021, Cryptnox.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 4.1.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../../../_sources/autoapi/cryptnoxpy/exceptions/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>